### PR TITLE
Magento to dynamics365 - Products and Prices

### LINKED ISSUE #93: Magento to Dynamics 365 - Products & Prices
Sync product information (name, description, price, SKU, and inventory) from Magento to Dynamics 365.

We have a Dynamics 365 connector (Request)

1. Use schedule trigger
2. Get product information from Magento
3. Go over products and create those in Dynamics 365. Update if those already exist.
4. On output state how many created, updated and if any errors occurred.

Example API Calls:
```http
GET /rest/V1/products
Host: your-magento-store.com
Authorization: Bearer your-magento-access-token
```

```http
POST /api/data/v9.1/products
Host: your-dynamics-instance.api.crm.dynamics.com
Authorization: Bearer your-dynamics-access-token
Content-Type: application/json

{
  "name": "Example Product",
  "description": "Example product description",
  "productnumber": "example-sku",
  "price": 29.99,
  "quantityonhand": 100
}
```

### PR DESCRIPTION
**Pull request by jannejjj**
closes #93 

Checklist for review:
- [x] Process name is fine, uses normal casing and descriptive
- [x] Process description is fine, uses normal casing, dots and is descriptive
  - [x] Element names use normal casing, no unneeded capital letters.
- [x] Correct casing used for formats and terms, e.g. CSV, FTP, JSON, Avro
  - [x] Element names do not contain dots
  - [x] Comments / note element text are in good language and end with dots.
- [x] Template parameters checked
  - [x] Parameter descriptions are good and end with dots.
  - [x] Sensitive parameters are marked as secrets
- [x] "Turn off logging" switch is on for any task that has sensitive data as output
- [x] End "balls" like error throws and return statements have titles (double-click) and reasonable information.
  - [x] Results have good summaries, preferably with statistics of some sort like "processed X files"
  - [x] Errors have some error info, usually from task (e.g. some ErrorMessage property contains additional error info)
- [x] Process is organised with groups (dashed boxes)
- [x] Straight lines are preferred where possible
- [x] Subprocesses are forbidden
- [x] Text mode is preferred over Expressions in expressions, e.g. instead of *$"This is {#var.error}"* use *This is {{#var.error}}*
- [x] Process metadata file is present and content is fine (version number and name)
- [x] Process export file is present (process.json)
- [x] Process description file is present (all sections filled in)
- [x] Process description has a SVG screenshot of the process
- [x] Tempalte files are going to the correct folder

## Summary by CodeRabbit

- **New Features**
  - Introduced a template for syncing product and price information between Magento and Dynamics365, including implementation details and error handling.
  - Added metadata for the Magento to Dynamics365 product and price synchronization process. 

- **Chores**
  - Adjusted JSON formatting in the metadata file for synchronizing customers from Magento to Salesforce.

### CODE REVIEW

File: Templates/E-Commerce/Magento/Synchronize customers from Magento to Salesforce/metadata.json
Comment (jannejjj) [inline Templates/E-Commerce/Magento/Synchronize customers from Magento to Salesforce/metadata.json:1]: Noticed there was an unnecessary whitespace here so removed it

**Review by FrendsSarlinS — CHANGES_REQUESTED** [review]
I'm all in for this! Just a few things I noticed:

1) The process variables naming convention thing from before, change URL to Url and ID to Id.

2) Most of the error returns after checking if the product exists return only the #result from the previous task. You might have overlooked those. Include a meaningful error message and/or ErrorMessage from the task, if it's sufficient.

3) The bot's suggestions for the long-description seem valid to me, see if you can apply those somehow!

**Comment by jannejjj** [comment]
Thanks for the comments! I'll change the variable names and the returns - I'm pretty sure i already did those but might've lost the changes at some point 🤔 I'll adjust the description a bit as well.

**Comment by jannejjj** [comment]
Updated!

**Review by FrendsSarlinS — APPROVED** [review]
Thanks for the fixes, it's all fine by me now!

File: Templates/E-Commerce/Magento/Magento to Dynamics365 - Products & Prices/long-description.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+This template process retrieves information about products and their prices from **Magento** and either inserts or updates them into **Dynamics365**.
+Using this template, you can synchronize products and their prices from Magento to Dynamics365.
```
Comment (FrendsSarlinS) [inline Templates/E-Commerce/Magento/Magento to Dynamics365 - Products & Prices/long-description.md:2]: Should we add a newline here?

**Pull request merged**